### PR TITLE
Video captions

### LINKED ISSUE #2867: Allow videos to have captions (same as e.g. image caption)
Videos can't currently take captions in the same way e.g. images can. They should be able to.

screen_shot_2018-06-20_at_09_44_35

### PR DESCRIPTION
**Pull request by davidpmccormick**
Closes #2867.

Allowing the editorial team to add captions below video (in the same way as is currently possible for e.g. images).

This PR pulls the caption out of `CaptionedImage` and replaces it with a `Caption` that is shared across `CaptionedImage`s and `VideoEmbed`s.

screen shot 2018-06-20 at 15 56 50

### CODE REVIEW

File: common/styles/utilities/compiled_variables/_fonts.scss [hunk 1]
```diff
@@ -25,7 +25,7 @@ $fonts: (
     'font-family-subset': "'Wellcome Bold Web Subset'",
     'font-family-web': "'Wellcome Bold Web'",
     'font-size': 33px,
-    'line-height': 48px
+    'line-height': 42px
```
Comment (davidpmccormick) [inline common/styles/utilities/compiled_variables/_fonts.scss]: This belonged in the typography PR from a couple of days ago, but slipped in here.

File: common/views/components/Caption/Caption.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+const Caption = ({caption, preCaptionNode}: Props) => {
+  return (
+    <figcaption className={`caption flex-inline font-black plain-text ${font({s: 'LR3', m: 'LR2'})} ${spacing({s: 3}, {padding: ['top', 'bottom']})}`}>
+      <Icon name='image' extraClasses='float-l margin-right-s1' />
+      <div className={`overflow-hidden ${spacing({s: 3, m: 4, l: 5}, {padding: ['right']})}`}
+        tabIndex='0'>
+        {preCaptionNode}
+        <PrismicHtmlBlock html={caption} />
+        <style>
+          {'.caption p { display: inline; }'}
```
Comment (davidpmccormick) [inline common/views/components/Caption/Caption.js]: ¯\_(ツ)_/¯

File: client/js/components/truncate-text.js [hunk 3]
```diff
@@ -1,72 +0,0 @@
...
-          } else {
-            truncateControl.classList.add('is-hidden');
-          }
-        });
-      }
-    });
-  }
-};
-
-export default truncateText;
```
Comment (jamesgorrie) [inline client/js/components/truncate-text.js]: 💯

File: common/views/components/Caption/Caption.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  preCaptionNode?: Node
+|}
+
+const Caption = ({caption, preCaptionNode}: Props) => {
+  return (
+    <figcaption className={`caption flex-inline font-black plain-text ${font({s: 'LR3', m: 'LR2'})} ${spacing({s: 3}, {padding: ['top', 'bottom']})}`}>
+      <Icon name='image' extraClasses='float-l margin-right-s1' />
+      <div className={`overflow-hidden ${spacing({s: 3, m: 4, l: 5}, {padding: ['right']})}`}
+        tabIndex='0'>
+        {preCaptionNode}
```
Comment (jamesgorrie) [inline common/views/components/Caption/Caption.js]: Could we allow this to only take an `Icon`?

File: server/services/prismic-parsers.js [hunk 5]
```diff
@@ -399,17 +399,19 @@ export function asText(maybeContent: any) {
 }
 
 export function asHtml(maybeContent: any) {
-  // Prismic can send us empty html elements which can lead to unwanted UI in templates.
-  // Check that `asText` wouldn't return an empty string.
-  const isEmpty = !maybeContent || asText(maybeContent).trim() === '';
-
-  return isEmpty ? null : RichText.asHtml(maybeContent, linkResolver).trim();
+  return isMissingOrEmpty(maybeContent) ? null : RichText.asHtml(maybeContent, linkResolver).trim();
```
Comment (jamesgorrie) [inline server/services/prismic-parsers.js]: Could we use this?
https://github.com/wellcometrust/wellcomecollection.org/blob/master/common/services/prismic/parsers.js#L57

Either here, or at the implementation?

File: common/views/components/MainMedia/MainMedia.js [hunk 6]
```diff
@@ -33,7 +33,11 @@ export const MainMedia = ({
   return (
     <figure className={`captioned-image ${extraClasses}`}>
       <div className='captioned-image__image-container'>
+        {/* https://github.com/facebook/flow/issues/2405 */}
```
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply common/views/components/MainMedia/MainMedia.js]: Turns out it has been, fixed it.

File: common/views/components/Caption/Caption.js (hunk 2 above)
Comment (jamesgorrie) [reply common/views/components/Caption/Caption.js]: https://github.com/wellcometrust/wellcomecollection.org/blob/master/common/styles/utilities/_root-scope-classes.scss#L422-L430

?

File: common/views/components/Caption/Caption.js (hunk 4 above)
Comment (davidpmccormick) [reply common/views/components/Caption/Caption.js]: the `preCaptionNode` here is for numbering images in e.g. a gallery (1/10). We'd still need to find another way to get that in.

File: server/services/prismic-parsers.js (hunk 5 above)
Comment (davidpmccormick) [reply server/services/prismic-parsers.js]: We spoke. Added `parseRichText` to the common parsers.

File: common/views/components/Caption/Caption.js (hunk 2 above)
Comment (davidpmccormick) [reply common/views/components/Caption/Caption.js]: oh yeah, maybe. Although I don't know if there's another instance that we'd want the first para from Prismic to display inline – feels kinda one-offy. I'm easy either way.

File: common/views/components/MainMedia/MainMedia.js [hunk 7]
```diff
@@ -33,11 +33,7 @@ export const MainMedia = ({
   return (
     <figure className={`captioned-image ${extraClasses}`}>
       <div className='captioned-image__image-container'>
-        {/* https://github.com/facebook/flow/issues/2405 */}
```
Comment (davidpmccormick) [inline common/views/components/MainMedia/MainMedia.js]: Ah – was failing for me locally; I should update my Flow.

Comment (jamesgorrie) [reply common/views/components/MainMedia/MainMedia.js]: You can use the one in `node_modules` to ensure consistency.

**Pull request merged**